ibis/pandas/core.py

```diff
@@ -324,6 +381,18 @@ def main_execute(expr, params=None, scope=None, aggcontext=None, **kwargs):
     if scope is None:
         scope = {}
 
+    if timecontext is None:
+        timecontext = {}
+    else:
+        try:
+            timecontext = tuple(map(pd.to_datetime, timecontext))
+        except ValueError:
+            raise com.IbisError(
```

Comment:
I would add a comment explaining what you are doing

Reply:
Same as above, move all checks to ```canonicalize_context()``` and documented.